core/server/data/migrations/versions/4.0/17-update-mobiledoc.js

```diff
@@ -0,0 +1,53 @@
+const logging = require('../../../../../shared/logging');
+const {createIrreversibleMigration} = require('../../utils');
+
+module.exports = createIrreversibleMigration(async (knex) => {
+    logging.info('Adding version to posts.mobiledoc objects and cleaning up deprecated card names');
+
+    await knex.transaction(async (trx) => {
+        // get list of posts ids, use .forUpdate to lock rows until the transaction is finished
+        const postIdRows = await knex('posts')
+            .transacting(trx)
+            .forUpdate()
+            .select('id');
```

Comment:
I opted for grabbing all post ids rather than id+mobiledoc because some sites have 1000's of posts and 100's of megabytes of post content which doesn't seem like a good candidate for pre-loading into memory.

Interested to hear if anyone things this is optimising for edge cases at the cost of slower migrations for the majority.

Reply:
This is the best approach imo 👍